config/routes.rb

```diff
@@ -149,6 +149,12 @@
     end
   end
 
+  namespace :api do
+    get 'generic_works/:id' => 'generic_works#show'
+    get 'generic_works/:id/add_parent/:parent_id' => 'generic_works#add_parent'
```

Comment:
I'm not sure custom routes just for adding a single parent are the best route. CC already has support for just sending a PATCH to the resource's controller with in_work_ids as a parameter - why not use that?